.github/workflows/publish.yml

```diff
@@ -0,0 +1,37 @@
+name: Publish Package to nuget
+
+on:
+  release:
+    types: [create]
+    
+jobs:
+  build:
+
+    env:
+      BUILD_CONFIG: 'Release'
+      SOLUTION: 'intacct-sdk-net.sln'
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Setup NuGet
+      uses: NuGet/setup-nuget@v1.0.5
+
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v2
+      with:
+        dotnet-version: 6.x
+        
+    - name: Restore dependencies
+      run: dotnet restore
+
+    - name: Build
+      run: dotnet build $SOLUTION --configuration $BUILD_CONFIG --no-restore
```

Comment:
$varName is fine because it's within the scope of the job/build. If the env was declared outside of the job/build, then ${{env.varName}} would be the right way :)